### PR TITLE
enhance and fix for softnet_data parser

### PR DESCRIPTION
**Pull request by remijouannet**
Hello,

enhance and fix for softnet_data parser
* the offset for cpu_collision, received_rps, flow_limit_count was not correct
*
https://github.com/torvalds/linux/commit/7d58e6555870d01d85b181ba2a16b217a1ea8bdd introduce two new columns, softnet_backlog_len and index
* the code example in the note was wrong, is_packet_drops is a proprety not a function
* additional documentation and links about softnet_data structure and history
* additional unit tests

### All Pull Requests:

Check all that apply:

* [Y] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [N] Is this PR to correct an issue?
* [Y] Is this PR an enhancement?

### Complete Description of Additions/Changes:

*Add your description here*

### CODE REVIEW

File: insights/parsers/softnet_stat.py [hunk 1]
```diff
@@ -8,42 +8,87 @@
...
-    Column-08: cpu_collision: collision occur while obtaining device lock while transmitting.
-    Column-09: received_rps: number of times cpu woken up received_rps.
-    Column-10: flow_limit_count: number of times reached flow limit count.
+    Column-09: cpu_collision: collision occur while obtaining device lock while transmitting.
+    Column-10: received_rps: number of times cpu woken up received_rps.
+    Column-11: flow_limit_count: number of times reached flow limit count.
+    Column-12: softnet_backlog_len: Backlog status
+    Column-13: index: core id owning this softnet_data
+
+For the proc file format details,
```
Comment (xiangce) [inline insights/parsers/softnet_stat.py:31]: The following docstring is not satisfied with the requirement of `sphinx`, and breaks the `doc-test`. Would you mind fixing them accordingly?
The insights-core follows the `Google Style Python Docstrings` please check [the examples here](https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html#example-google) for reference

File: insights/parsers/softnet_stat.py [hunk 2]
```diff
@@ -69,9 +114,11 @@ def parse_content(self, contents):
             self.cpu_nstats.append({'packet_process': stats[0],
                             'packet_drops': stats[1],
                             'time_squeeze': stats[2],
-                            'cpu_collision': stats[7],
-                            'received_rps': stats[8],
-                            'flow_limit_count': stats[9]})
+                            'cpu_collision': stats[8],
```
Comment (xiangce) [inline insights/parsers/softnet_stat.py:117]: It looks like the new columns were introduced to the kernel on `Sep 22, 2020`, I think there must be some servers still using the old version without the new columns. Could you please update the code to make it compatible with both the old and new versions?

File: insights/tests/parsers/test_softnet_stat.py [hunk 3]
```diff
@@ -9,9 +9,27 @@
 """
 
 SOFTNET_STAT_2 = """
-00008e78 00000000 000000011 00000000 00000000 00000000 00000000 00000000 00000020 00000010
-000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 000000010 00000001
+00008e78 00000000 00000011 00000000 00000000 00000000 00000000 00000000 00000020 00000010
+000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000010 00000001
```
Comment (xiangce) [inline insights/tests/parsers/test_softnet_stat.py:13]: Thanks for fixing these tests.

**Review by xiangce — CHANGES_REQUESTED** [review]
Hey @remijouannet,  Thank you very much for contributing to insights-core. Besides the following comments, Could you please also fix the `flake8` errors to make the pipeline pass? 
Please check the flake8 errors [here](https://github.com/RedHatInsights/insights-core/actions/runs/3291223040/jobs/5425728869#step:5:79)

Thanks.

File: insights/parsers/softnet_stat.py (hunk 1 above)
Comment (remijouannet) [reply insights/parsers/softnet_stat.py:31]: should be good now

File: insights/parsers/softnet_stat.py [hunk 4]
```diff
@@ -8,42 +8,58 @@
...
             change in short term.
 
 Examples:
     >>> SOFTNET_STATS = '''
-    ... 00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
-    ... 000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
-    ... 0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
-    ... 0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
-    ...'''.strip()
+    ... 00358fe3 00006283 00000000 00000000 00000000 00000000 00000000 00000000 00000000 000855fc 00000000 00000000 00000000
```
Comment (ryan-blakley) [inline insights/parsers/softnet_stat.py:50]: If you're going to update the examples, could you please add in a doc test and update the examples to work properly? The below commit is an example of me adding a doc test to existing parser for reference.

https://github.com/RedHatInsights/insights-core/commit/cbf6b7bdf8ceb0712553d2165f2ff860e0c849e1

**Review by ryan-blakley — CHANGES_REQUESTED** [review]

File: insights/parsers/softnet_stat.py (hunk 2 above)
Comment (remijouannet) [reply insights/parsers/softnet_stat.py]: the code now handle 9, 10, 11 or 13 columns, if key doesn't have any value you get None

File: insights/parsers/softnet_stat.py (hunk 4 above)
Comment (ryan-blakley) [reply insights/parsers/softnet_stat.py]: @remijouannet It doesn't look like you've added a doctest yet, was this marked resolved by accident?

Comment (remijouannet) [reply insights/parsers/softnet_stat.py]: sorry i forgot to add a unitest 

```
bin/pytest -v insights/tests/parsers/test_softnet_stat.py
=================================================================== test session starts ====================================================================
platform darwin -- Python 3.9.12, pytest-7.1.3, pluggy-1.0.0 -- /Users/remi.jouannet/git/insights-core/bin/python3.9
cachedir: .pytest_cache
rootdir: /Users/remi.jouannet/git/insights-core, configfile: setup.cfg
plugins: cov-4.0.0
collected 2 items

insights/tests/parsers/test_softnet_stat.py::test_softnet_stat PASSED                                                                                [ 50%]
insights/tests/parsers/test_softnet_stat.py::test_softnet_stat_doc_examples PASSED
```
better ?

Comment (ryan-blakley) [reply insights/parsers/softnet_stat.py]: Yep that's great thank you.

File: insights/parsers/softnet_stat.py (hunk 2 above)
Comment (xiangce) [reply insights/parsers/softnet_stat.py]: Thanks for the update, it looks good to me now.

**Review by ryan-blakley — APPROVED** [review]
Looks good to me.

**Review by xiangce — APPROVED** [review]

**Pull request merged**